harness/determined/pytorch/_callback.py

```diff
@@ -39,6 +41,14 @@ def on_train_step_end(self, step_id: int, metrics: Dict[str, Any]) -> None:
         """
         pass
 
+    def on_before_optimizer_step(self, parameters: Generator) -> None:
```

Comment:
Can we rename this `on_optimizer_step_start()` to be consistent with e.g. `on_validation_step_start()` or `before_optimizer_step()` to be simple?

`on_before_optimizer_step()` sounds super awkward.

Reply:
IMO, naming patterns are helpful in bringing clarity, until they make things less clear.  And I think this is a place where following the pattern brings less clarity.

I won't block this change over this though.